iothub/service/src/Messaging/FileUploadNotificationProcessorClient.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Common.Extensions;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> for handling file upload notications.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-file-upload#service-file-upload-notifications"/>.
+    public class FileUploadNotificationProcessorClient : IDisposable
+    {
+        private readonly string _hostName;
+        private readonly IotHubConnectionProperties _credentialProvider;
+        private readonly IotHubConnection _connection;
+
+        /// <summary>
+        /// The callback to be executed each time file upload notification is received from the service.
+        /// </summary>
+        /// <remarks>
+        /// May not be null.
+        /// </remarks>
+        public Func<FileNotification, DeliveryAcknowledgement> _fileNotificationProcessor;
+
+        /// <summary>
+        /// The callback to be executed when the connection is lost.
+        /// </summary>
+        /// <remarks>
+        /// May not be null.
+        /// </remarks>
+        public Action<Exception> _errorProcessor;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected FileUploadNotificationProcessorClient()
+        {
+        }
+
+        internal FileUploadNotificationProcessorClient(
+            string hostName,
+            IotHubConnectionProperties credentialProvider,
+            IotHubServiceClientOptions options)
+        {
+            _hostName = hostName;
+            _credentialProvider = credentialProvider;
+            _connection = new IotHubConnection(credentialProvider, options.UseWebSocketOnly, options.TransportSettings, options);
+            FileNotificationReceiver = new AmqpFileNotificationReceiver(_connection);
+        }
+
+        /// <summary>
+        /// Gets the AmqpFileNotificationReceiver which can deliver notifications for file upload operations.
+        /// </summary>
+        internal AmqpFileNotificationReceiver FileNotificationReceiver { get; }
+
+        /// <summary>
+        /// Open the AmqpFileNotificationReceiver instance.
```

Comment:
Customers have no idea what an ```AmqpFileNotificationReceiver``` is.

```suggestion
        /// Open the connection and start receiving file upload notifications.
```